### PR TITLE
Performance improvements

### PR DESCRIPTION
**Pull request by eoinoconnell**
- Use dedicated queue for socket communication and parsing
- Ensure parsing HTML is completed on correct thread. We show a loader on conversation view when getting initial messages so parsing is complete at that stage.
- Only parse HTML when dealing with messages for the current conversation and not on socket updates or getting list of conversations
- Pin star scream version to fix build issues
- Removed unused fields and classes
- Fix issue where reloading a failed message would not mark it as sent correctly.

### CODE REVIEW

File: Drift/Managers/SocketManager.swift [hunk 1]
```diff
@@ -83,24 +84,32 @@ class SocketManager {
     }
     
     func computeShardId(orgId: Int) -> Int{
-        return orgId % 50 //WS_NUM_SHARDS
+        return orgId % 50
     }
     
     func willReconnect() {
-        NotificationCenter.default.post(name: .driftSocketStatusUpdated, object: self, userInfo: ["connectionStatus": ConnectionStatus.connecting])
+        DispatchQueue.main.async {
```
Comment (timMorrill) [inline Drift/Managers/SocketManager.swift:91]: Is the idea that these notifications update the UI and need to be on the main thread?

**Review by timMorrill — APPROVED** [review]
Just a question about the threading for the notifications, but we can chat about that tomorrow

File: Drift/Managers/SocketManager.swift (hunk 1 above)
Comment (eoinoconnell) [reply Drift/Managers/SocketManager.swift:91]: Yeah exactly, my expectation would be that NSNotifications would always be on the main treat but happy to do it differently if you have another option?

Comment (timMorrill) [reply Drift/Managers/SocketManager.swift:91]: I was just curious

**Review by timMorrill — APPROVED** [review]

**Pull request merged**